app/src/main/java/agency/five/codebase/android/movieapp/ui/home/HomeViewModel.kt

```diff
@@ -0,0 +1,118 @@
+package agency.five.codebase.android.movieapp.ui.home
+
+import agency.five.codebase.android.movieapp.data.repository.MovieRepository
+import agency.five.codebase.android.movieapp.model.MovieCategory
+import agency.five.codebase.android.movieapp.ui.home.mapper.HomeScreenMapper
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.flow.*
+import kotlinx.coroutines.launch
+
+class HomeViewModel(
+    private val movieRepository: MovieRepository,
+    homeScreenMapper: HomeScreenMapper,
+) : ViewModel() {
+    private val initialPopularMoviesViewState =
+        movieRepository.popularMovies(MovieCategory.POPULAR_STREAMING)
+            .map { movies ->
+                homeScreenMapper.toHomeMovieCategoryViewState(
+                    listOf(
+                        MovieCategory.POPULAR_STREAMING,
+                        MovieCategory.POPULAR_ONTV,
+                        MovieCategory.POPULAR_FORRENT,
+                        MovieCategory.POPULAR_INTHREATRES
+                    ),
+                    MovieCategory.POPULAR_STREAMING, movies
+                )
+            }
+
+    private val _selectedPopularCategoryId: MutableStateFlow<Int> = MutableStateFlow(0)
+
+    private val _popularMoviesViewState: MutableStateFlow<HomeMovieCategoryViewState> =
+        MutableStateFlow(HomeMovieCategoryViewState(emptyList(), emptyList()))
+
+    val popularMoviesViewState: StateFlow<HomeMovieCategoryViewState> = combine(
+        initialPopularMoviesViewState,
+        _selectedPopularCategoryId
+    ) { initialPopularMoviesViewState, selectedCategoryId ->
+        val movieCategories = initialPopularMoviesViewState.movieCategories.map {
+            it.copy(isSelected = selectedCategoryId == it.itemId)
+        }
+        initialPopularMoviesViewState.copy(movieCategories = movieCategories)
+    }.stateIn(viewModelScope, SharingStarted.Eagerly, _popularMoviesViewState.value)
```

Comment:
Using `combine` will trigger this block if either of the two flows emit a new value. But here we want to update viewstate only if `_selectedPopularCategoryId` got new value and then `initialPopularMoviesViewState` got new values. This can be achieved by using `.flatMapLatest {}`on `_selectedPopularCategoryId` and pass in `initialPopularMoviesViewState`. If you do that you don't even need to create `initialPopularMoviesViewState`, and can instead just pass the inner code here.